python/lsst/daf/butler/cli/opt/collection_type.py

```diff
@@ -0,0 +1,58 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import click
+
+from lsst.daf.butler.registry import CollectionType
+
+
+class collection_type_option:  # noqa: N801
+    """Decorator to add a collection type option to a click command.
+
+    Converts the type option from string to a CollectionType enum value.
+    """
+
+    defaultHelp = "If provided, only list collections of this type."
+
+    choices = ["CHAINED", "RUN", "TAGGED"]
+
+    def makeCollectionType(self, context, param, value):
+        if value is None:
+            return value
+        value = value.upper()
```

Comment:
Click doesn't normalize case on choice options if you say `case_sensitive=False` ? Does click do minimum matching so that "chain" would work as well as "chained"?